envi/archs/arm/disasm.py

```diff
@@ -2395,14 +2405,19 @@ def getOperValue(self, op, emu=None):
         return reglist
 
     def repr(self, op):
+            #fixed register list. Should be {r1, r2, r3 ..} not { r1 r2 r3 ..}
+            cnt = 0
             s = [ "{" ]
             for l in xrange(16):
                 if (self.val & (1<<l)):
+                    cnt += 1
+                    if cnt > 1:
+                        s.append(', ')
                     s.append(arm_regs[l][0])
             s.append('}')
             if self.oflags & OF_UM:
                 s.append('^')
-            return " ".join(s)
+            return "".join(s)
```

Comment:
you're working too hard.

make the register list a separate list
', '.join(regs)

should do what you want.

then insert this string into the other list of strings and return ''.join(s).
less python, go faster.  in fact, cut out the whole "for l" loop and use something like this:

regs = [ arg_regs[l][0] for l in range(16) if (self.val & (1<<l)) ]
s.append( ','.join(regs) )

then, cnt goes away as well and we're left with like 7 lines total.  wash, rinse, repeat for render.